contrib/ubase/patches/respawn.patch

```diff
@@ -0,0 +1,88 @@
+Index: ubase/ubase/respawn.c
+===================================================================
+--- ubase.orig/ubase/respawn.c	2022-12-31 13:21:26.743034241 +0100
++++ ubase/ubase/respawn.c	2022-12-31 13:22:17.386365903 +0100
+@@ -6,7 +6,6 @@
+ 
+ #include <errno.h>
+ #include <fcntl.h>
+-#include <poll.h>
+ #include <signal.h>
+ #include <stdio.h>
+ #include <stdlib.h>
+@@ -32,22 +31,14 @@
+ int
+ main(int argc, char *argv[])
+ {
+-	char *fifo = NULL;
+ 	unsigned int delay = 0;
+ 	pid_t pid;
+-	char buf[BUFSIZ];
+ 	int savederrno;
+-	ssize_t n;
+-	struct pollfd pollset[1];
+-	int polln;
+ 
+ 	ARGBEGIN {
+ 	case 'd':
+ 		delay = estrtol(EARGF(usage()), 0);
+ 		break;
+-	case 'l':
+-		fifo = EARGF(usage());
+-		break;
+ 	default:
+ 		usage();
+ 	} ARGEND;
+@@ -55,42 +46,11 @@
+ 	if (argc < 1)
+ 		usage();
+ 
+-	if (fifo && delay > 0)
+-		usage();
+-
+ 	setsid();
+ 
+ 	signal(SIGTERM, sigterm);
+ 
+-	if (fifo) {
+-		pollset->fd = open(fifo, O_RDONLY | O_NONBLOCK);
+-		if (pollset->fd < 0)
+-			eprintf("open %s:", fifo);
+-		pollset->events = POLLIN;
+-	}
+-
+ 	while (1) {
+-		if (fifo) {
+-			pollset->revents = 0;
+-			polln = poll(pollset, 1, -1);
```

Comment:
As a challenge please consider implementing `poll` emulation in similar way `select` was added here https://github.com/cahirwpz/mimiker/blob/master/lib/libc/sys/select.c

Reply:
Is it something you'd like to commit to?

Reply:
This has potentially a big impact since `poll` functionality is used in wide variety of programs.

Reply:
> Is it something you'd like to commit to?

For now, I'd rather start dealing with `pathconf` and `TIOCNOTTY` ioctl (https://github.com/cahirwpz/mimiker/pull/1390#issuecomment-1596098488) and then try to implement `poll`.

Reply:
Since `pathconf` and `TIOCNOTTY` PRs have already been merged, we're left only with `poll` I guess.

Reply:
Since `poll` has been recently implemented, I restored almost entire `respawn.c` original source code (https://github.com/cahirwpz/mimiker/pull/1371/commits/e05e66377e665e827ebd198fcdd7b4fb95c44860).